### ISSUE TITLE
Typescript 3 compilation error in index.d.ts

### ISSUE THREAD

**Issue by rodrigoelp** [react-navigation/react-navigation-redux-helpers#61]
Hi, I've started using this dependency in a brand new project and off the bat I get a compilation error from here as follows:

```sh
master ✱ ❯ tsc -v
Version 3.0.1

master ✱ ❯ tsc
node_modules/react-navigation-redux-helpers/index.d.ts:25:71 - error TS2344: Type 'S' does not satisfy the constraint 'Action<any>'.

25   ): React.ComponentType<{ state: NavigationState; dispatch: Dispatch<S> }>;
                                                                         ~
```

Do you guys know what the problem might be?

I've setup my project with the following dependencies:

```json
{
  "dependencies": {
    "react": "16.4.1",
    "react-native": "0.56.0",
    "react-navigation": "2.11.2",
    "react-navigation-redux-helpers": "2.0.5",
    "react-redux": "5.0.7",
    "redux": "4.0.0",
    "redux-thunk": "2.3.0"
  },
  "devDependencies": {
    "@types/react": "16.4.9",
    "@types/react-native": "0.56.7",
    "@types/react-navigation": "2.0.15",
    "@types/react-redux": "6.0.6",
    "babel-jest": "23.4.2",
    "babel-preset-react-native": "^5",
    "jest": "23.5.0",
    "react-test-renderer": "16.4.1",
    "typescript": "3.0.1"
  },
}
```

**Comment by rodrigoelp** [comment]
To be more explicit, the may problem is that dispatch should take an `A` which extends of `Action` and in here, this definition forces an `S` of state as an action (which is the reason why the compiler is complaining)...

Has anybody used this type definition successfully before?

**Comment by Ashoat** [comment]
Hi @rodrigoelp, I unfortunately don’t maintain the Typescript bindings for this project and rely on PRs from the community to keep it updated. I’d suggest looking through the history of the libdef file to find folks who might be able to answer your questions.

**Comment by rodrigoelp** [comment]
Or, if you want ashoat. I can submit a pull request fixing it. I've checked the bindings for v2.9 and the compilation error is reported there as well (you just need to turn on the proper flags to evaluate your typings)

**Comment by Ashoat** [comment]
PRs are welcome!

**Comment by rodrigoelp** [comment]
Hi, sorry for the delay... Here is the PR.

**Comment by rodrigoelp** [comment]
Feel free to close this issue once it has been merged.

**Comment by Ashoat** [comment]
Pro-tip: you can set up a PR to auto-close an issue by just including the text "Fixes #61" in the text of a commit.
